workflow/controller/operator.go

```diff
@@ -230,7 +230,11 @@ func (woc *wfOperationCtx) operate(ctx context.Context) {
 		woc.updated = wfUpdate
 		if !acquired {
 			woc.log.Warn("Workflow processing has been postponed due to concurrency limit")
-			woc.wf.Status.Message = msg
+			phase := woc.wf.Status.Phase
+			if phase == wfv1.WorkflowUnknown {
```

Comment:
@sarabala1979 Is saw strange behaviour here. The workflow had completed (`phase=Failed`) but `acquired=false`. Why did we try and acquire a lock for a completed workflow?

Reply:
I don't understand how this could ever happen. Before operate() is invoked, we check to see if the workflows is completed and return. This should be impossible.

Reply:
it could be failed workflow retry.